src/main/java/scmspain/karyon/restrouter/core/ParamTypeResolver.java

```diff
@@ -16,11 +16,11 @@ public T resolveValueType(Class<T> classType, String value){
     } else if (classType.isAssignableFrom(boolean.class) || classType.isAssignableFrom(Boolean.class)) {
       return (T)Boolean.valueOf(value);
     } else if (classType.isAssignableFrom(long.class) || classType.isAssignableFrom(Long.class)) {
-      return (T)Boolean.valueOf(value);
+      return (T)Long.valueOf(value);
     } else if (classType.isAssignableFrom(short.class) || classType.isAssignableFrom(Short.class)) {
-      return (T)Boolean.valueOf(value);
+      return (T)Short.valueOf(value);
     } else if (classType.isAssignableFrom(float.class) || classType.isAssignableFrom(Float.class)) {
-      return (T)Boolean.valueOf(value);
+      return (T)Float.valueOf(value);
     }
     return null;
```

Comment:
I've been trying to improve that if-else block, don't know if what I finally got is better but, I think it's the way to go:

```java
public class ParamTypeResolver<T> {

  Map<Class, Function<String, T>> resolver = new HashMap<>();

  ParamTypeResolver() {
    resolver.put(int.class, (val) -> (T) Integer.valueOf(val));
    resolver.put(double.class, (val) -> (T) Double.valueOf(val));
    resolver.put(boolean.class, (val) -> (T) Boolean.valueOf(val));
    resolver.put(long.class, (val) -> (T) Long.valueOf(val));
    resolver.put(short.class, (val) -> (T) Short.valueOf(val));
    resolver.put(float.class, (val) -> (T) Float.valueOf(val));

    resolver.put(String.class, (val) -> (T) val);
    resolver.put(Integer.class, (val) -> (T) Integer.valueOf(val));
    resolver.put(Double.class, (val) -> (T) Double.valueOf(val));
    resolver.put(Boolean.class, (val) -> (T) Boolean.valueOf(val));
    resolver.put(Long.class, (val) -> (T) Long.valueOf(val));
    resolver.put(Short.class, (val) -> (T) Short.valueOf(val));
    resolver.put(Float.class, (val) -> (T) Float.valueOf(val));
  }

  public T resolveValueType(Class<T> classType, String value){

    if (resolver.containsKey(classType)) {
      return resolver.get(classType).apply(value);
    }

    return null;
  }

}
```